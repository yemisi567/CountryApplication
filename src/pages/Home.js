import SearchBox from "../components/SearchBox";
import CountryCards from "../components/CountryCards";
import { useState, useEffect } from "react";
import { useDispatch } from "react-redux";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { getCountries, getRegion } from "../redux/actions/countries";

Enzyme.configure({ adapter: new Adapter() });

const Home = () => {
  const dispatch = useDispatch();
  const [loading, setLoading] = useState(false);
  const [data, setData] = useState([]);
  const [query, setQuery] = useState("");

  const getCountriesData = async () => {
    try {
      const res = await dispatch(getCountries());
      res && setData(res.payload);
      res && setLoading(false);
    } catch (err) {
      console.log(err);
    }
  };

  const getRegionData = async (value) => {
    try {
      const res = await dispatch(getRegion(value));
      res && setData(res.payload);
      res && setLoading(false);
    } catch (err) {
      console.log(err);
    }
  };

  useEffect(() => {
    setLoading(true);
    const timer = setTimeout(() => {
      getCountriesData();
    }, 500);
    return () => clearTimeout(timer);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleOnChange = (value) => {
    setLoading(true);
    if (value === "all") {
      getCountriesData();
    } else {
      const timer = setTimeout(() => {
        getRegionData(value);
      }, 500);
      return () => clearTimeout(timer);
    }
  };

  const handleInput = (value) => {
    "input";
    setQuery(value);
  };

  return (
    <div className="pt-28">
      <SearchBox
        handleOnChange={handleOnChange}
        handleInput={handleInput}
      />
      <CountryCards
        setData={setData}
        data={data}
        status={data?.status}
        loading={loading}
        query={query}
      />
    </div>
  );
};

export default Home;
