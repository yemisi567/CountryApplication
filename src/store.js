import { configureStore } from "@reduxjs/toolkit";
import countriesReducer from "./redux/reducers/reducer";

export const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
});

export default store;
