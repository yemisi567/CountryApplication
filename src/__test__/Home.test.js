import React from "react";
import Home from "../pages/Home";
import { Provider } from "react-redux";
import { shallow } from "enzyme";
import store from "../store";

describe("Home Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Provider store={store}>
        <Home />
      </Provider>
    );
  });

  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
