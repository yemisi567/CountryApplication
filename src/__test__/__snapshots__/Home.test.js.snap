// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders properly 1`] = `ShallowWrapper {}`;

exports[`Home Component should render properly 1`] = `ShallowWrapper {}`;

exports[`Home should render correctly 1`] = `
<DocumentFragment>
  <div
    class="pt-28"
  >
    <div
      class="md:flex justify-between md:pt-14 md:pb-10 md:gap-10 px-8 md:px-10 "
    >
      <div
        class="flex  rounded-md items-center lg:w-[45vw] gap-4 mt-10 mb-6 md:mt-0 md:mb-0 dark:bg-[#2b3945]  bg-white shadow-md w-full shadow-gray-100 dark:shadow-gray-800 px-6 py-4"
      >
        <span />
        <input
          class="text-sm outline-none w-full bg-transparent dark:text-white"
          placeholder="Search for a country ..."
          type="text"
        />
      </div>
      <select
        class="p-4 flex justify-start text-md outline-none dark:bg-[#2b3945] dark:text-white"
        data-testid="countries"
        id="countries"
        name="countries"
      >
        <option
          disabled=""
          selected=""
          value="selected"
        >
          Filter by Region
        </option>
        <option
          value="africa"
        >
          Africa
        </option>
        <option
          value="america"
        >
          America
        </option>
        <option
          value="asia"
        >
          Asia
        </option>
        <option
          value="europe"
        >
          Europe
        </option>
        <option
          value="oceania"
        >
          Oceania
        </option>
        <option
          value="all"
        >
          All
        </option>
      </select>
    </div>
    <div>
      <div
        class="grid pb-10 xl:grid-cols-4 md:grid-cols-2 px-8 justify-items-center dark:bg-[#1F2D36] bg-[#FAFAFA]"
        data-testid="query"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home should render the component 1`] = `ShallowWrapper {}`;

exports[`Home should render the home page 1`] = `ShallowWrapper {}`;
